packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -15,6 +15,18 @@ import { TagOptions } from '../tag-options';
 import { hashValues } from './util';
 import { InputValidator } from './validation';
 
+interface LaunchRoleProp {
+  readonly roleArn: string,
+  readonly localRoleName?: never,
+}
+
+interface LaunchRoleNameProp {
+  readonly localRoleName: string,
+  readonly roleArn?: never,
+}
+
+type LaunchRoleConstraintRoleProp = LaunchRoleProp | LaunchRoleNameProp;
```

Comment:
These definitions I would actually move _below_ where they are first used (reason being that these are private to this file).

Reply:
Also, the naming convention for these is "Options" - "Props" are only used for construct properties. (For example, it's `CommonConstraintOptions`, not `CommonConstraintProps`)

Reply:
i think `type` and `interface` need to be outside of `class` but I moved it below